packages/vscode-extension/src/project/deviceSession.ts

```diff
@@ -287,12 +287,23 @@ export class DeviceSession implements Disposable {
   }
 
   public async changeDeviceSettings(settings: DeviceSettings): Promise<boolean> {
+    const didOnlyReplaySettingsChanged =
+      JSON.stringify(this.deviceSettings) ===
+      JSON.stringify({ ...settings, replaysEnabled: !settings.replaysEnabled });
+
     this.deviceSettings = settings;
+
     if (settings.replaysEnabled && !this.isLaunching) {
       this.device.startReplays();
     } else {
       this.device.stopReplays();
     }
+
+    if (didOnlyReplaySettingsChanged) {
```

Comment:
I looked up implementation of changeSettings and it seem like it only returns `true` when the parameter in question changes. So the only thing that this changes is the problem where we can trigger some commands during boot that the device doesn't like. 

Given that's the case, I don't think the introduced changes address the root cause of that, because even if we can enable replays during boot, changing any other device setting during that time would still result in that error.

I think this has been discussed in #590 already and the consensus was to block device settings entirely for the duration of boot which would also elliminate the need for this change.

Reply:
You're correct, disabling all device settings during boot is better solution, so I've added it in update.